h2/src/main/org/h2/expression/Function.java

```diff
@@ -1494,6 +1495,49 @@ private Value getValueWithArgs(Session session, Expression[] args) {
         case DATE_DIFF:
             result = ValueLong.get(datediff(v0.getString(), v1, v2));
             break;
+        case DATE_TRUNC:   
+            // Retrieve the field value (e.g. 'day', 'microseconds', etc.).
+            String fieldValue = v0.getString().toLowerCase();
+
+            // Retrieve the dateValue.
+            long[] fieldDateAndTime = DateTimeUtils.dateAndTimeFromValue(v1);
+            long fieldDateValue = fieldDateAndTime[0];
+
+            // Case where the date has to be truncated to the day.
+            if (fieldValue.equals("day")) {
+               
+                if (v1 instanceof ValueTime) {
+
+                    // Case where we truncate the time to the day (e.g.
+                    // '10:00:00 -> 00:00:00).
+                    result = v1.subtract(v1);
```

Comment:
`ValueTime.fromNanos(0)` should be used to get time at start of the day, but actually PostgreSQL returns `INTERVAL` in this case, not the `TIME` value. H2 doesn't have `INTERVAL` now, so it's may be better to return `TIMESTAMP` here.